lunatik_core.c

```diff
@@ -194,21 +196,15 @@ static int lunatik_newruntime(lunatik_object_t **pruntime, lua_State *parent, co
 	lunatik_toruntime(L) = runtime;
 	runtime->private = L;
 
-	lunatik_setversion(L);
-	luaL_openlibs(L);
-	if (sleep) {
-		luaL_requiref(L, "lunatik", luaopen_lunatik, 0);
-		lua_pop(L, 1); /* lunatik library */
-	}
-
-	if (lunatik_dofile(L, script) != LUA_OK) {
-		lunatik_runerror(L, parent, lua_tostring(L, -1));
+	lua_pushcfunction(L, lunatik_runscript);
+	lua_pushlightuserdata(L, (void *)script);
```

Comment:
not a factual suggestion for this review, per se, but rather, just to mention: about all these type casts, I really like those `cast_*` macros Lua define in `lua/llimits.h`; maybe Lunatik could have their own.

Reply:
we could use it if you think it's worth.. I wouldn't replicate them though.. we could include this header on lunatik.h perhaps..

Reply:
but I would apply this to the code as a whole, not only in this single place now.. what do you think?

Reply:
I just mentioned as something to consider for future changes.

Reply:
btw, you can create a issue for that..